src/platform/mhv/tests/hooks/useMyHealthAccessGuard.unit.spec.jsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { expect } from 'chai';
+import sinon from 'sinon';
+import { waitFor } from '@testing-library/react';
+
+import { renderWithStoreAndRouter } from '@department-of-veterans-affairs/platform-testing/react-testing-library-helpers';
+import { useMyHealthAccessGuard } from '../../hooks/useMyHealthAccessGuard';
+
+const stateFn = ({ mhvAccountState = 'OK' } = {}) => ({
+  user: {
+    profile: {
+      mhvAccountState,
+    },
+  },
+});
+
+const Component = () => {
+  useMyHealthAccessGuard();
+  return <h1>Health Tool</h1>;
+};
+
+const setup = (initialState = stateFn()) =>
+  renderWithStoreAndRouter(<Component />, {
+    initialState,
+  });
+
+describe('useMyHealthAccessGuard', () => {
+  let originalLocation;
```

Comment:
Is this safe? Do you really need to set the original location at the end? I feel you can just set the spy before the test and nothing else.

Reply:
This is a fairly common pattern in `vets-website` for spying on `window.location.replace`.